internal/documentmap/documentmap.go

```diff
@@ -103,23 +105,17 @@ func (m *Map) ImportFromCursor(ctx context.Context, cursor *mongo.Cursor) error
 	for cursor.Next(ctx) {
 		err := cursor.Err()
 		if err != nil {
-			if err == mongo.ErrNoDocuments {
+			if errors.Is(err, mongo.ErrNoDocuments) {
 				break
 			}
 
 			return err
 		}
 
 		nDocumentsReturned++
+		bytesReturned += (int64)(len(cursor.Current))
 
-		var rawDoc bson.Raw
-		err = cursor.Decode(&rawDoc)
-		if err != nil {
-			return err
-		}
-		bytesReturned += (int64)(len(rawDoc))
-
-		m.addDocument(rawDoc)
+		m.addDocument(cursor.Current)
```

Comment:
nit: Since the copy in `addDocument` is now crucial to correctness, I suggest renaming it `copyAndAddDocument` or `addCopyOfDocument`.